lib/active_model_serializers/adapter/json_api/resource_identifier.rb

```diff
@@ -2,11 +2,30 @@ module ActiveModelSerializers
   module Adapter
     class JsonApi
       class ResourceIdentifier
+        def self.type_for(class_name, serializer_type = nil, transform_options = {})
+          if serializer_type
+            raw_type = serializer_type
+          else
+            inflection =
+              if ActiveModelSerializers.config.jsonapi_resource_type == :singular
```

Comment:
I feel like this could be extracted to a method somewhere. I could maybe see this logic being used elsewhere